.gitlab-ci.yml

```diff
@@ -0,0 +1,138 @@
+stages:
+  - install_deps
+  - lint
+  - test
+  - build
+  - deploy
+
+variables:                         &default-vars
+  CI_IMAGE:                        "node:16.10"
+  APP_PORT:                        3000
+  DEPLOYMENT_IMAGE_REGISTRY:       docker.io/paritytech/pr-custom-review
+  DEPLOYMENT_TAG:                  $CI_COMMIT_SHORT_SHA-$CI_PIPELINE_ID
+  # Vault Section
+  VAULT_ADDR:                      https://vault.parity-mgmt-vault.parity.io
+  VAULT_AUTH_PATH:                 gitlab-parity-io-jwt
+  VAULT_AUTH_ROLE:                 cicd_gitlab_parity_${CI_PROJECT_NAME}
+  HELM_SECRETS_DRIVER:             vals
+
+default:
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+      - unknown_failure
+      - api_failure
+  interruptible:                   true
+  tags:
+    - kubernetes-parity-build
+  image:                           $CI_IMAGE
+
+.common-refs:                      &common-refs
+  rules:
+    - if: $CI_COMMIT_REF_NAME == "master"
+    - if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/                 # PRs
+
+.staging-refs:                     &staging-refs
+  rules:
+    - if: $CI_COMMIT_REF_NAME =~ /^stg-v[0-9]+\.[0-9]+.*$/  # e.g. stg-v0.1
```

Comment:
`$CI_COMMIT_TAG` would be more precise here but it doesn't make too much of a difference

Reply:
a1c5b7a8538bf04bf023f8f7da56c74a377f0520